GameServer/Logic/GameObjects/Pathfinder.cs

```diff
@@ -107,22 +119,26 @@ public static Path getPath(Vector2 from, Vector2 to, float boxSize)
             }
             return getPath(from, to, PATH_DEFAULT_BOX_SIZE(getMesh().getSize()));
         }
+
         public static void setMap(Map map)// { chart = map; mesh = chart->getAIMesh(); }
         {
             chart = map;
         }
+
         public static AIMesh getMesh()
         {
-            if (chart == null)
+            if (chart != null)
             {
-                _logger.LogCoreError("The map hasn't been set but the mesh was requested.");
-                return null;
+                return chart.AIMesh;
             }
-            return chart.AIMesh;
+
+            _logger.LogCoreError("The map hasn't been set but the mesh was requested.");
+            return null;
         }
+
         private static float PATH_DEFAULT_BOX_SIZE(float map_size)
         {
-            return map_size / (float)GRID_SIZE;
+            return map_size / GRID_SIZE;
```

Comment:
I'm not sure if `GRID_SIZE` is a float or not by default, but if it isn't, don't remove the cast. I'm assuming it is since removing the cast would be smart then.